front-end/src/components/main-page/MainPage.jsx

```diff
@@ -17,8 +20,24 @@ class MainPage extends Component {
     this.setState({showTranscriptUploadModal: !this.state.showTranscriptUploadModal});
   }
 
-  openAnalysisPage = () => {
-    store.dispatch(openAnalysisPage());
+  openAnalysisPage = async() => {
+    try {
+      await this.getAnalyzedData();
+      store.dispatch(openAnalysisPage());
+    } catch (e) {
+      window.alert("Error in analyzing transcript. " + e.response.data.error);
+    }
+  }
+
+  getAnalyzedData = async() =>{
+      const override = store.getState().analyzeTranscript.override;
+      const projetId = store.getState().analyzeTranscript.projectIdToBeAnalyzed;
+      const transcripts = store.getState().analyzeTranscript.analyzedTranscripts;
+      if(override || (!(projetId in transcripts))){
```

Comment:
Can you explain what the 2 cases this if-statement represent are? I know one represents if the transcript has been overridden, but I'm not sure what the other means

Reply:
So these are some extra code for when(if) we want to handle multiple transcripts and project ids. The analyzed transcripts is an object mapping each project id to the analyzed transcript so that if the user wants to travel between pages and look at different projects it doesn't have to call backend if it already has been analyzed and stored in redux. I will also add some comments to clarify

Reply:
+1 for type of projetId to projectId